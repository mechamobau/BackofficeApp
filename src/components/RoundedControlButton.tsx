import React from 'react';
import type {ReactText} from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import useTheme from '../hooks/useTheme';

type RoundedControlButtonProps = TouchableOpacityProps & {
  children: ReactText;
};

function RoundedControlButton({
  children,
  ...touchableOpacityProps
}: RoundedControlButtonProps) {
  const {theme} = useTheme();

  return (
    <TouchableOpacity
      style={[styles.controlButton, {borderColor: theme.buttonBorderColor}]}
      {...touchableOpacityProps}
    >
      <Text style={[styles.controlButtonText, {color: theme.textColor}]}>
        {children}
      </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  controlButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 1,
  },
  controlButtonText: {
    fontSize: 22,
  },
});

export default RoundedControlButton;
