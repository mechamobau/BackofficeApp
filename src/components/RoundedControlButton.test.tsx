import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import RoundedControlButton from './RoundedControlButton';
import {ReactTestInstance} from 'react-test-renderer';

describe('RoundedControlButton | component | integration test', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(
      <RoundedControlButton>This is a button</RoundedControlButton>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders the children inside a button label text', () => {
    const {queryByText, rerender} = render(
      <RoundedControlButton>This is a button</RoundedControlButton>,
    );

    expect(queryByText('This is a button')).toBeTruthy();

    rerender(
      <RoundedControlButton>This is a change children</RoundedControlButton>,
    );

    expect(queryByText('This is a button')).not.toBeTruthy();
  });
  it('should call onPress when button is pressed', () => {
    const onPressMock = jest.fn();

    const {queryByA11yRole} = render(
      <RoundedControlButton onPress={onPressMock} accessibilityRole="button">
        This is a button
      </RoundedControlButton>,
    );

    const button = queryByA11yRole('button');

    expect(button).toBeTruthy();

    fireEvent.press(button as ReactTestInstance);

    expect(onPressMock).toHaveBeenCalled();
  });
});
