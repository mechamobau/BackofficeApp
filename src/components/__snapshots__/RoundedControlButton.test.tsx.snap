// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedControlButton | component | integration test should renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#000000",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 40,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
  >
    This is a button
  </Text>
</View>
`;
